### ISSUE TITLE
Feedback: New User testing feedback from Heuristics test.

### ISSUE THREAD

**Issue by grantkemp** [google/google-api-objectivec-client-for-rest#207]
We did a heuristics testing exercise on this library as part of a workshop a few days ago and thought I would share the outputs: 
 
Here are some of the common issues that cropped up below for those getting started. 
Would be a great idea to help new users of this library understand: 
from the docs: 
- In the documentation it mentions that it will only work with Google's APIs which are JSON based. 
 Is there a list of the APIS that are JSON based? e.g. Google Analytics v4/ Drive / Sheets?
- What does the developer need to know to decide which of the following they need to use: 
Should they use:
a) this library only, 
or 
b) This library and toolbox for Mac: https://github.com/google/gtm-oauth2, 
or 
c)  GTMAppAuth - https://github.com/google/GTMAppAuth and when should they use

**Comment by thomasvl** [comment]
Formatting is a little odd, here's so I'm not completely sure what you are after.

> Which of the APIS are JSON based e.g. Google Analytics v4/ Drive / Sheets?

The main readme mentions this, all the services in the [Source/GeneratedServices directory](https://github.com/google/google-api-objectivec-client-for-rest/tree/master/Source/GeneratedServices) are supported.

The rest is where the formatting gets a little strange and I'm not completely sure what you are asking.

**Comment by grantkemp** [comment]
@thomasvl  - thanks, I fixed the formatting ( that will teach me to edit GH on a mobile!) 

Basically, the feedback we had was that there are 3 potential libraries to interact with Google's APIs.  It would be useful to have a single place to help understand in what circumstances should a developer use what library.

**Comment by thomasvl** [comment]
Did you look at the main Readme and its pointers to the wiki? Information about authentication/authorization (gtm-oauth2 and GTMAppAuth) are talked about there.

**Comment by grantkemp** [comment]
Hi @thomasvl  
thanks for the brilliant feedback.  

We did look at it, however, I think the document for the wiki appears to try and cover A lot of use cases -without going into enough detail for each of them to allow a user to successfully complete them. 

E.g. I have just tried to do the user case of implementing the library in Swift using the Cocoapods implementation in Mac OSX.   In one part of the readme - it says that GTMAppAuth is included however if I look further down I need to depend on it.  

When I tried to do this - the project wouldn't build as it appears that GTMAppAuth doesn't work with latest MacOSX Xcode without generating errors.  

I don't mind spending the time to write a simple one pager for the Cocoapods/ swift/ Mac OSX journey that would list the steps from pod init to successfully accessing the data if I could get it to work eg: 

1. pod init - add the following:  
``` pod 'GoogleAPIClientForREST'
  pod 'GTMAppAuth'```
2. Open the new workspace. 
Unfortunately, that's as far as I could get as the project doesn't build. 
Even then - I am not sure what the next stages would be- I would expect to see 
1. Authorizaton Stage: 
a. Do ... 
b. Do.. 

2. Reading Data
a. 
b. 
c. 

If you would like to do more advanced topics such as writing then go here.

Hope that clarifies?
